### ISSUE TITLE
Improve executor default security posture

### ISSUE THREAD

**Issue by alexec** [argoproj/argo-workflows#7964]
The executor has a number of things we could improve:

- [x] #3961 
- [ ] #7955 
- [x] #7829 
- [x] #6895 (depends on #7804)
- [x] #7486 

---

**Message from the maintainers**:

Love this enhancement proposal? Give it a 👍. We prioritise the proposals with the most 👍.

**Comment by tooptoop4** [comment]
close as per https://blog.argoproj.io/practical-argo-workflows-hardening-dd8429acc1ce ?

**Comment by Joibel** [comment]
Closing this umbrella issue as superseded by #7955, which tracks the only remaining item.
